src/transaction/display.rs

```diff
@@ -0,0 +1,24 @@
+use crate::Transaction;
+
+use core::fmt;
+use core::str::FromStr;
+
+impl fmt::Display for Transaction {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // its not worthy to extend the API requirement of serde because the transaction
```

Comment:
what do you mean by worthy?

Reply:
True, its not clear. The idea behind that is the fact that serde implements all serializations as fallible because you might have some attribute that can't be serialized (maybe pointers?)

That is not the case for a transaction, so we can just safely ignore that. I rephrased the comments to clarify that.